modules/home-manager/waybar.nix

```diff
@@ -0,0 +1,47 @@
+{ config
+, options
+, lib
+, sources
+, ...
+}:
+let
+  cfg = config.programs.waybar.catppuccin;
+  enable = cfg.enable && config.programs.waybar.enable;
+in
+{
+  options.programs.waybar.catppuccin = (lib.ctp.mkCatppuccinOpt "waybar") // {
+    mode = lib.mkOption {
+      type = lib.types.enum [
+        "prependImport"
+        "createLink"
+      ];
+      default = "prependImport";
+      description = "Whether to prepend an `@import` (requires that `programs.waybar.style` is set to a string) or to create a symlink `~/.config/waybar/catppuccin.css` (requires you to import manually).";
+    };
+  };
+
+  config = lib.mkIf enable (
+    let
+      styleFile = "${sources.waybar}/themes/${cfg.flavour}.css";
+    in
+    {
+      warnings =
```

Comment:
i don't think this warning is needed

`lib.modules.defaultOverridePriority` is currently `100`, which is equal to `lib.mkBefore`'s priority. this makes it so that under the precondition of the `"prependImport"` mode being used, this will never trigger. `lib.mkBefore` can also be used freely in the final configuration, with those entries still being placed after/having a higher priority than ours.

Reply:
Unless someone uses `lib.mkForce` (or similar). Then this will also override the definition in the module.

Vice versa `lib.mkForce (lib.mkBefore "...")` would override any definition using the default override priority (or higher value than `mkForce`'s 50).

And something like `lib.mkOverride options.programs.waybar.style.highestPrio (...)` is not possible due to infinite recursion (and should not be used anyway, as it can't be overridden).

Reply:
In some sense this is just me thinking that `lib.mkBefore` does prepend something to an option (regardless of priority), but it actually is just a definition that should be merged first.^^

So I can perfectly understand if we want to remove it, since users should be aware that `lib.mkForce` overrides things.

Reply:
yeah to me this is intended behavior. if the only situation this really helps in is with the use of `mkForce`, i feel like that could go for all modules here (which we don't have warnings for)

Reply:
removed the warning